src/hooks/useAllV3Ticks.ts

```diff
@@ -0,0 +1,100 @@
+import { Token } from '@uniswap/sdk-core'
+import { FeeAmount, Pool, TICK_SPACINGS } from '@uniswap/v3-sdk'
+import { nearestUsableTick, TickMath } from '@uniswap/v3-sdk/dist/'
+import { ZERO_ADDRESS } from '../constants/misc'
+import { useEffect, useMemo, useState } from 'react'
+import { Result, useSingleContractMultipleData } from 'state/multicall/hooks'
+import { useTickLens } from './useContract'
+import { TickData } from 'constants/ticks'
+
+function bitmapIndex(tick: number, tickSpacing: number) {
+  return Math.floor(tick / tickSpacing / 256)
+}
+
+const REFRESH_FREQUENCY = { blocksPerFetch: 24 }
+
+// for now, reconsider using this function, it consumes a lot of data and cpu to fetch all the ticks.
```

Comment:
this is why this was removed

Reply:
Yeah, I have found perf to be acceptable (increased blocks per fetch 24, hogs the cpu for about .5-1s when processing the response) and data usage is around 300kb at 0.05% which seems reasonable given the usefulness of the data provided.

I left the comment just because it is fairly relevant.

We could consider getting the active liquidity data preprocessed by the graph to reduce both data usage and cpu.

Reply:
we should just use the graph, we already brought it into the app. the app freezing for 1s on a high end laptop when you get to a page is a pretty bad perf degradation

Reply:
It's closer to 400-500ms on a high-end laptop. There's a spinner while this is happening so user is aware that something is being loaded. 

We only fetch the blocks every 5 minutes or so. I think it should be reasonable to start.

Grabbing data from the graph has some other implications. Optimizing for data availability makes sense to start with I think.

Reply:
personally i'm in favor of fetching the data on-chain:

1) the trust assumptions are more minimal
2) the total bandwidth transferred is probably not _that_ different. we're obviously using more "cloud compute" with this approach, but acceptable given the setting
3) there are almost certainly perf gains still to be had in this PR/throughout the data fetching lifecycle, up to and including moving compute into a web worker, which imo is an important thing for us to start thinking about in general, in preparation for e.g. client-side routing

Reply:
Discussed offline. While the preference is to rely on chain data in general, there are some pain points that are difficult to work around with (the current implementation of) TickLens (e.g. active liquidity from old blocks).

Will move to The Graph.